src/io/file_descriptor.windows.cr

```diff
@@ -194,90 +144,44 @@ class IO::FileDescriptor
   end
 
   private def unbuffered_read(slice : Bytes)
-    count = slice.size
-    loop do
-      bytes_read = LibC.read(@fd, slice.pointer(count).as(Void*), count)
-      if bytes_read != -1
-        return bytes_read
-      end
-
-      if Errno.value == Errno::EAGAIN
-        wait_readable
-      else
-        raise Errno.new "Error reading file"
-      end
-    end
-  ensure
-    if (readers = @readers) && !readers.empty?
-      add_read_event
+    overlapped = GC.malloc(sizeof(LibWindows::Overlapped)).as(LibWindows::Overlapped*)
```

Comment:
You may use `Pointer(LibWindows::Overlapped).malloc` for a simpler allocation.

Maybe we can avoid the HEAP allocation altogether? We can create the struct on the stack, and pass a pointer to it, which will be valid for the duration of the call.

```crystal
overlapped = LibWindows::Overlapped.new # all members are set to 0
overlapped.offset = @pos

if LibWindows.read_file(..., out bytes_read, pointerof(overlapped))
  @pos += bytes_read
end
```